### DISCUSSION TITLE
posts/my-new-desktop/

### DISCUSSION THREAD

**Discussion by giscus** [Shitao5/shitao-blog#48] [Comments]
# posts/my-new-desktop/

近期俺的笔记本电脑越来越卡顿，蓝屏频率直线上升1，俺想着该给自己配台性能好些的主机了，也算「宴请儿时的自己」2。
[&hellip;] 想着 618 或许有优惠，五月就联系了做电脑生意的战友，他说六月初的价格已经和 618 差不多了。于是六月初定好配置单，具体如下：
[&hellip;] 暂时没有配显卡，一是根据俺使用核显笔记本的经验，或许 CPU 自带的核显就已足够；二是年底出 50 系显卡，等 &hellip;

https://shitao5.org/posts/my-new-desktop/

**Comment by CyrusYip** [comment]
> 截止目前发现大多数时候都在 WSL 里边耍。

不装个 Linux 耍耍？

**Reply by Shitao5** [reply]
WSL 里边是 Ubuntu 哦嘿嘿

**Reply by Shitao5** [reply]
准备给笔记本直接装 Ubuntu 24 尝尝鲜。

**Reply by CyrusYip** [reply]
我知道 WSL，我意思是装个真的 Linux 系统。看你新电脑硬盘空间那么大，可以随便装了。

**Reply by Shitao5** [reply]
新电脑暂时不装，WSL 带来的便利性很不错。旧的笔记本准备自己拆开清灰换个硅脂之后「装个真的 Linux 系统」了。

**Reply by CyrusYip** [reply]
我从 Windows 10 开始改用 Linux，在 Linux 写程序很舒服，基本上需要啥都是一条命令装好。其实我很好奇为啥经常用 WSL 的人不改用 Linux，正好可以请教你这个问题。是有些软件只有 Windows 版，还是觉得安装 Linux 麻烦，又或者是其他原因？

**Reply by Shitao5** [reply]
看来你还遇到过很多用 WSL 这样「死性不改」的家伙啊哈哈。我赞成 Linux 写程序很舒服，装 Linux 倒也不麻烦。我用 WSL 的原因更多是它结合了 Windows 和 Linux 的优点：

1. 对 Windows 的路径依赖，一直用着感觉也不赖。像 Microsoft（俺还是 365 会员），Adobe，微信这样的软件还是用在 Windows 里边顺手。
2. WSL 可以用 Linux 处理 Windows 里边的文件（性能会有一点小损失），也可以 Windows 资源管理器看到 Linux 的目录，这比双系统或者虚拟机省事多了。

**Reply by CyrusYip** [reply]
> 看来你还遇到过很多用 WSL 这样「死性不改」的家伙啊哈哈。

我在网上多次看到有人说「Windows 是最好的 Linux 发行版」（应该是想表达 WSL 很好用），这句话甚至出现于讨论 Linux 发行版选择的帖子。我（Linux 用户）认为这个说法超离谱。首先 Windows 不是 Linux 发行版；其次 WSL2 只是运行于虚拟机而且功能更弱的 Linux，并不能取代实机安装的 Linux。说这句话的人大概没用过真的 Linux 吧。

写到这里我发现自己对离谱言论还是太认真了，以后还是要忽略那些看着离谱又不加解释的言论。最近看到有人说 [年轻人已经不再尊重电脑](https://meta.appinn.net/t/topic/57719)，这标题成功骗到我进去看那篇[公众号文章](https://mp.weixin.qq.com/s/-CdWyLpB-276RDStcNL0-Q)，我一开始还在想怎么反驳，现在觉得回应这些标题党文章简直浪费时间。

我还真遇到有人给我强力推荐 WSL，我说我已经在用 Arch Linux 了，他还是继续强力推荐 WSL。我当时心想：我都用着 Linux 了，为啥还叫我去运行 Windows 然后在里面用 Linux 虚拟机。最后我还化郁闷为幽默，做了张迷因图：[A friend recommended WSL to me (based on my true experience) : r/linuxmemes](https://www.reddit.com/r/linuxmemes/comments/12ckj42/a_friend_recommended_wsl_to_me_based_on_my_true/)。

![arch-meme](https://github.com/Shitao5/shitao-blog/assets/60951091/76a41d04-2038-4ae3-895c-563e15a160a8)

---

回应你说的两点：

1. 对于 Linux 缺少 Windows 软件的问题，有几种解决方法
    1. 找替代品，比如用 WPS 代替 Microsoft Office
    2. 用 Windows 虚拟机运行软件
    3. 用 [Wine](https://www.winehq.org/) 运行软件，不需要运作整个 Windows 更省资源，但兼容性比虚拟机差
    4. 用双系统，重启到 Windows 用软件
    5. 在另一台电脑用 Windows（当你不拘泥于用一个系统或者一台电脑干所有事情，你会轻松很多）
    6. 补充一下，现在 Linux 也能装 [微信](https://flathub.org/apps/com.tencent.WeChat)（可以用小程序）
2. Linux 可以挂载 Windows 使用的 NTFS 分区。我不知道 Windows 能不能挂载 Linux 使用的分区格式（ext4、btrfs 等）。

**Reply by Shitao5** [reply]
Windows 也能挂载 Linux 使用的分区：

![俺工作电脑的WSL](https://github.com/Shitao5/shitao-blog/assets/68451957/e7446282-e80b-415e-ba85-751ee50202e2)

这些解决方法我大致知道，作为用户，我更追求软件使用的体验，软件运行在什么硬件、什么操作系统并不重要。对我而言，Microsoft Office 就是比 WPS 好用（Microsoft 365 更是优于 Microsoft 2019 这些），Windows 也会比虚拟机，Wine 带来更好的体验。适合自己的才是最好的，[论坛](https://meta.appinn.net/t/topic/57719/5)里有一个评论说「我为什么要去尊重一个工具？」，爱惜工具但不被其捆绑住是最好的状态。

WSL 是给离不开 Windows 但又有 Linux 使用需求的用户（比如我）一个非常好的解决方案。对于你这样的 Linux 老玩家，用 WSL 完全是舍近求远，可以想到你的无语哈哈。

> 当你不拘泥于用一个系统或者一台电脑干所有事情，你会轻松很多

不拘泥于一个操作系统确实会轻松很多，但是在硬件上，我反而觉得能合并更好，因为大部分情况主力机只有一台，还省去文件传输的麻烦。比如我要用剪映剪视频（字幕识别、音频生成、智能口播等功能很方便，暂时没有替代软件），但它只支持 Windows 和 Mac 两个系统，Windows + WSL 这样的方案就很适合我。

我和你正好是反着的，你是 Linux + 运行 Windows 软件的方法或另一台 Windows 电脑，我是 Windows + WSL 以及另一台运行 Linux 的电脑（俺笔记本已经装好 Ubuntu 24.04 了）哈哈。

**Reply by Liam-Zhong** [reply]
有打游戏的需求且有两台电脑，当然是旧电脑物理装Linux😁

**Comment by Amo-nick** [comment]
记住，不要折腾linux发行版

**Reply by Shitao5** [reply]
年轻还是可以多折腾，说不定能打开新世界的大门呢😉
